INFO:root:/tmp/vis/plot.json
INFO:root:/tmp/vis
INFO:root:/tmp/vis/plot.json
INFO:root:/tmp/vis/plot.json
INFO:root:/tmp/vis
INFO:root:/tmp/vis/plot.json
INFO:root:/tmp/vis/plot.json
INFO:root:/tmp/vis
INFO:root:modified : /tmp/vis/plot.json
INFO:root:closed : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis
INFO:root:deleted : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis
INFO:root:created : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis
INFO:root:modified : /tmp/vis/plot.json
INFO:root:closed : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 24, in handle_connect
    handler = Handler()
TypeError: __init__() missing 1 required positional argument: 'socketio'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 535, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 24, in handle_connect
    handler = Handler()
TypeError: __init__() missing 1 required positional argument: 'socketio'
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:
INFO:root:modified : /tmp/vis/plot.json
INFO:root:
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:<class 'str'>
INFO:root:
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 24, in handle_connect
    handler = Handler(socketio)
TypeError: Handler() takes no arguments

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 535, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 24, in handle_connect
    handler = Handler(socketio)
TypeError: Handler() takes no arguments
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 24, in handle_connect
    handler = Handler(socketio)
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 34, in __init__
    self.update(data)
UnboundLocalError: local variable 'data' referenced before assignment
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/nb.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:/tmp/vis/plot.json
INFO:root:/tmp/vis/plot.json
INFO:root:/tmp/vis/plot.json
INFO:root:/tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json: {"data":[{"alignmentgroup":"True","hovertemplate":"Year=%{x}<br>Cards Released=%{y}<extra></extra>","legendgroup":"","marker":{"color":"#636efa","pattern":{"shape":""}},"name":"","offsetgroup":"","orientation":"v","showlegend":false,"textposition":"auto","x":[1993,1994,1995,1996,1997,1998,1999,2000,2001,2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022,2023,2024],"xaxis":"x","y":[1596,1643,1728,1020,1607,1162,1639,918,1592,826,1610,958,2136,1122,1497,1310,1540,1799,2114,1483,2100,2807,2762,2906,3530,3794,4938,6771,8009,12275,10325,124],"yaxis":"y","type":"bar"}],"layout":{"template":{"data":{"bar":[{"error_x":{"color":"#2a3f5f"},"error_y":{"color":"#2a3f5f"},"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"bar"}],"barpolar":[{"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"barpolar"}],"carpet":[{"aaxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"baxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"type":"carpet"}],"choropleth":[{"colorbar":{"outlinewidth":0,"ticks":""},"type":"choropleth"}],"contour":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"contour"}],"contourcarpet":[{"colorbar":{"outlinewidth":0,"ticks":""},"type":"contourcarpet"}],"heatmap":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"heatmap"}],"heatmapgl":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"heatmapgl"}],"histogram":[{"marker":{"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"histogram"}],"histogram2d":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"histogram2d"}],"histogram2dcontour":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"histogram2dcontour"}],"mesh3d":[{"colorbar":{"outlinewidth":0,"ticks":""},"type":"mesh3d"}],"parcoords":[{"line":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"parcoords"}],"pie":[{"automargin":true,"type":"pie"}],"scatter":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatter"}],"scatter3d":[{"line":{"colorbar":{"outlinewidth":0,"ticks":""}},"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatter3d"}],"scattercarpet":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scattercarpet"}],"scattergeo":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scattergeo"}],"scattergl":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scattergl"}],"scattermapbox":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scattermapbox"}],"scatterpolar":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatterpolar"}],"scatterpolargl":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatterpolargl"}],"scatterternary":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatterternary"}],"surface":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"surface"}],"table":[{"cells":{"fill":{"color":"#EBF0F8"},"line":{"color":"white"}},"header":{"fill":{"color":"#C8D4E3"},"line":{"color":"white"}},"type":"table"}]},"layout":{"annotationdefaults":{"arrowcolor":"#2a3f5f","arrowhead":0,"arrowwidth":1},"autotypenumbers":"strict","coloraxis":{"colorbar":{"outlinewidth":0,"ticks":""}},"colorscale":{"diverging":[[0,"#8e0152"],[0.1,"#c51b7d"],[0.2,"#de77ae"],[0.3,"#f1b6da"],[0.4,"#fde0ef"],[0.5,"#f7f7f7"],[0.6,"#e6f5d0"],[0.7,"#b8e186"],[0.8,"#7fbc41"],[0.9,"#4d9221"],[1,"#276419"]],"sequential":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"sequentialminus":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]},"colorway":["#636efa","#EF553B","#00cc96","#ab63fa","#FFA15A","#19d3f3","#FF6692","#B6E880","#FF97FF","#FECB52"],"font":{"color":"#2a3f5f"},"geo":{"bgcolor":"white","lakecolor":"white","landcolor":"#E5ECF6","showlakes":true,"showland":true,"subunitcolor":"white"},"hoverlabel":{"align":"left"},"hovermode":"closest","mapbox":{"style":"light"},"paper_bgcolor":"white","plot_bgcolor":"#E5ECF6","polar":{"angularaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"bgcolor":"#E5ECF6","radialaxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"scene":{"xaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","gridwidth":2,"linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white"},"yaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","gridwidth":2,"linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white"},"zaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","gridwidth":2,"linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white"}},"shapedefaults":{"line":{"color":"#2a3f5f"}},"ternary":{"aaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"baxis":{"gridcolor":"white","linecolor":"white","ticks":""},"bgcolor":"#E5ECF6","caxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"title":{"x":0.05},"xaxis":{"automargin":true,"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","zerolinewidth":2},"yaxis":{"automargin":true,"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","zerolinewidth":2}}},"xaxis":{"anchor":"y","domain":[0.0,1.0],"title":{"text":"Year"},"tickmode":"linear","tick0":1,"dtick":1},"yaxis":{"anchor":"x","domain":[0.0,1.0],"title":{"text":"Cards Released"}},"legend":{"tracegroupgap":0},"margin":{"t":60},"barmode":"relative","title":{"text":"Cards Released by Year","x":0.5,"y":0.98}}}
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json: {"data":[{"alignmentgroup":"True","hovertemplate":"color=White<br>Week=%{x}<br>Number of Cards=%{y}<extra></extra>","legendgroup":"White","marker":{"color":"tan","pattern":{"shape":""}},"name":"White","offsetgroup":"White","orientation":"v","showlegend":true,"textposition":"auto","x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53],"xaxis":"x","y":[81,118,120,239,486,177,228,43,72,59,173,57,400,72,169,513,414,395,118,3,44,247,472,276,283,110,228,480,538,150,453,116,193,323,71,390,67,122,446,889,351,164,6,202,455,549,194,367,236,31,27,58,24],"yaxis":"y","type":"bar"},{"alignmentgroup":"True","hovertemplate":"color=Blue<br>Week=%{x}<br>Number of Cards=%{y}<extra></extra>","legendgroup":"Blue","marker":{"color":"blue","pattern":{"shape":""}},"name":"Blue","offsetgroup":"Blue","orientation":"v","showlegend":true,"textposition":"auto","x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53],"xaxis":"x","y":[86,133,128,233,407,183,222,36,48,49,188,55,388,83,141,447,385,327,116,3,46,246,460,261,268,105,242,431,536,150,412,173,214,263,41,347,67,111,474,848,355,149,40,184,460,532,201,358,262,22,26,69,28],"yaxis":"y","type":"bar"},{"alignmentgroup":"True","hovertemplate":"color=Black<br>Week=%{x}<br>Number of Cards=%{y}<extra></extra>","legendgroup":"Black","marker":{"color":"black","pattern":{"shape":""}},"name":"Black","offsetgroup":"Black","orientation":"v","showlegend":true,"textposition":"auto","x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53],"xaxis":"x","y":[74,137,127,251,422,192,185,41,36,52,229,58,362,77,184,420,395,346,125,2,45,249,491,273,254,101,243,449,583,151,421,130,147,374,84,364,73,139,470,955,286,173,6,226,432,576,164,412,290,31,24,124,31],"yaxis":"y","type":"bar"},{"alignmentgroup":"True","hovertemplate":"color=Red<br>Week=%{x}<br>Number of Cards=%{y}<extra></extra>","legendgroup":"Red","marker":{"color":"red","pattern":{"shape":""}},"name":"Red","offsetgroup":"Red","orientation":"v","showlegend":true,"textposition":"auto","x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53],"xaxis":"x","y":[79,135,135,236,403,181,219,18,42,47,178,60,446,87,136,465,371,333,99,2,40,262,519,287,252,114,224,454,570,152,443,118,170,283,52,370,72,84,448,871,322,163,33,218,444,565,178,402,283,25,23,76,31],"yaxis":"y","type":"bar"},{"alignmentgroup":"True","hovertemplate":"color=Green<br>Week=%{x}<br>Number of Cards=%{y}<extra></extra>","legendgroup":"Green","marker":{"color":"green","pattern":{"shape":""}},"name":"Green","offsetgroup":"Green","orientation":"v","showlegend":true,"textposition":"auto","x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53],"xaxis":"x","y":[99,139,132,244,450,175,233,18,39,53,179,48,377,109,149,460,417,340,124,45,296,537,300,290,103,230,460,556,153,458,148,196,371,99,364,66,128,481,888,290,173,5,264,485,546,237,394,303,28,25,83,31],"yaxis":"y","type":"bar"},{"alignmentgroup":"True","hovertemplate":"color=Colorless<br>Week=%{x}<br>Number of Cards=%{y}<extra></extra>","legendgroup":"Colorless","marker":{"color":"grey","pattern":{"shape":""}},"name":"Colorless","offsetgroup":"Colorless","orientation":"v","showlegend":true,"textposition":"auto","x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53],"xaxis":"x","y":[130,169,133,78,303,149,151,14,104,48,93,26,342,65,89,424,219,249,106,71,187,455,295,322,76,161,245,327,89,541,210,137,176,86,315,50,55,424,814,265,135,7,137,305,690,194,322,240,32,27,122,19],"yaxis":"y","type":"bar"},{"alignmentgroup":"True","hovertemplate":"color=Multicolor<br>Week=%{x}<br>Number of Cards=%{y}<extra></extra>","legendgroup":"Multicolor","marker":{"color":"gold","pattern":{"shape":""}},"name":"Multicolor","offsetgroup":"Multicolor","orientation":"v","showlegend":true,"textposition":"auto","x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53],"xaxis":"x","y":[38,144,95,400,567,191,303,39,90,52,247,55,114,60,34,858,922,747,182,2,62,200,540,380,335,57,476,163,325,142,242,189,115,322,76,733,74,180,440,1196,629,60,5,428,410,740,236,162,129,20,82,51,36],"yaxis":"y","type":"bar"}],"layout":{"template":{"data":{"bar":[{"error_x":{"color":"#2a3f5f"},"error_y":{"color":"#2a3f5f"},"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"bar"}],"barpolar":[{"marker":{"line":{"color":"#E5ECF6","width":0.5},"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"barpolar"}],"carpet":[{"aaxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"baxis":{"endlinecolor":"#2a3f5f","gridcolor":"white","linecolor":"white","minorgridcolor":"white","startlinecolor":"#2a3f5f"},"type":"carpet"}],"choropleth":[{"colorbar":{"outlinewidth":0,"ticks":""},"type":"choropleth"}],"contour":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"contour"}],"contourcarpet":[{"colorbar":{"outlinewidth":0,"ticks":""},"type":"contourcarpet"}],"heatmap":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"heatmap"}],"heatmapgl":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"heatmapgl"}],"histogram":[{"marker":{"pattern":{"fillmode":"overlay","size":10,"solidity":0.2}},"type":"histogram"}],"histogram2d":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"histogram2d"}],"histogram2dcontour":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"histogram2dcontour"}],"mesh3d":[{"colorbar":{"outlinewidth":0,"ticks":""},"type":"mesh3d"}],"parcoords":[{"line":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"parcoords"}],"pie":[{"automargin":true,"type":"pie"}],"scatter":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatter"}],"scatter3d":[{"line":{"colorbar":{"outlinewidth":0,"ticks":""}},"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatter3d"}],"scattercarpet":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scattercarpet"}],"scattergeo":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scattergeo"}],"scattergl":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scattergl"}],"scattermapbox":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scattermapbox"}],"scatterpolar":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatterpolar"}],"scatterpolargl":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatterpolargl"}],"scatterternary":[{"marker":{"colorbar":{"outlinewidth":0,"ticks":""}},"type":"scatterternary"}],"surface":[{"colorbar":{"outlinewidth":0,"ticks":""},"colorscale":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"type":"surface"}],"table":[{"cells":{"fill":{"color":"#EBF0F8"},"line":{"color":"white"}},"header":{"fill":{"color":"#C8D4E3"},"line":{"color":"white"}},"type":"table"}]},"layout":{"annotationdefaults":{"arrowcolor":"#2a3f5f","arrowhead":0,"arrowwidth":1},"autotypenumbers":"strict","coloraxis":{"colorbar":{"outlinewidth":0,"ticks":""}},"colorscale":{"diverging":[[0,"#8e0152"],[0.1,"#c51b7d"],[0.2,"#de77ae"],[0.3,"#f1b6da"],[0.4,"#fde0ef"],[0.5,"#f7f7f7"],[0.6,"#e6f5d0"],[0.7,"#b8e186"],[0.8,"#7fbc41"],[0.9,"#4d9221"],[1,"#276419"]],"sequential":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]],"sequentialminus":[[0.0,"#0d0887"],[0.1111111111111111,"#46039f"],[0.2222222222222222,"#7201a8"],[0.3333333333333333,"#9c179e"],[0.4444444444444444,"#bd3786"],[0.5555555555555556,"#d8576b"],[0.6666666666666666,"#ed7953"],[0.7777777777777778,"#fb9f3a"],[0.8888888888888888,"#fdca26"],[1.0,"#f0f921"]]},"colorway":["#636efa","#EF553B","#00cc96","#ab63fa","#FFA15A","#19d3f3","#FF6692","#B6E880","#FF97FF","#FECB52"],"font":{"color":"#2a3f5f"},"geo":{"bgcolor":"white","lakecolor":"white","landcolor":"#E5ECF6","showlakes":true,"showland":true,"subunitcolor":"white"},"hoverlabel":{"align":"left"},"hovermode":"closest","mapbox":{"style":"light"},"paper_bgcolor":"white","plot_bgcolor":"#E5ECF6","polar":{"angularaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"bgcolor":"#E5ECF6","radialaxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"scene":{"xaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","gridwidth":2,"linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white"},"yaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","gridwidth":2,"linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white"},"zaxis":{"backgroundcolor":"#E5ECF6","gridcolor":"white","gridwidth":2,"linecolor":"white","showbackground":true,"ticks":"","zerolinecolor":"white"}},"shapedefaults":{"line":{"color":"#2a3f5f"}},"ternary":{"aaxis":{"gridcolor":"white","linecolor":"white","ticks":""},"baxis":{"gridcolor":"white","linecolor":"white","ticks":""},"bgcolor":"#E5ECF6","caxis":{"gridcolor":"white","linecolor":"white","ticks":""}},"title":{"x":0.05},"xaxis":{"automargin":true,"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","zerolinewidth":2},"yaxis":{"automargin":true,"gridcolor":"white","linecolor":"white","ticks":"","title":{"standoff":15},"zerolinecolor":"white","zerolinewidth":2}}},"xaxis":{"anchor":"y","domain":[0.0,1.0],"title":{"text":"Week"}},"yaxis":{"anchor":"x","domain":[0.0,1.0],"title":{"text":"Number of Cards"}},"legend":{"title":{"text":"color"},"tracegroupgap":0},"title":{"text":"Cards Released by Color by week"},"barmode":"relative"}}
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 25, in handle_connect
    handler = Handler(socketio, app)
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 39, in __init__
    self.update(init_path)
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 49, in update
    self.socketio.emit('update', self.read_file(path), broadcast=True)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 462, in emit
    self.server.emit(event, *args, namespace=namespace, to=to,
TypeError: emit() got an unexpected keyword argument 'broadcast'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 535, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 25, in handle_connect
    handler = Handler(socketio, app)
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 39, in __init__
    self.update(init_path)
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 49, in update
    self.socketio.emit('update', self.read_file(path), broadcast=True)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 462, in emit
    self.server.emit(event, *args, namespace=namespace, to=to,
TypeError: emit() got an unexpected keyword argument 'broadcast'
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 25, in handle_connect
    target=check_update,
NameError: name 'check_update' is not defined
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:root:modified : /tmp/vis/plot.json
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 24, in handle_connect
    self.socketio.start_background_task(
AttributeError: 'NoneType' object has no attribute 'socketio'
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 27, in handle_connect
    self.socketio.start_background_task(
AttributeError: 'NoneType' object has no attribute 'socketio'
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 20, in handle_connect
    file=app.config['file'],
KeyError: 'file'
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 19, in handle_connect
    ensure_directory(app)
NameError: name 'ensure_directory' is not defined
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
ERROR:engineio.server:message handler error
Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 823, in _handle_event
    ret = handler(auth)
TypeError: handle_connect() takes 0 positional arguments but 1 was given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/n/anaconda3/lib/python3.8/site-packages/engineio/server.py", line 431, in run_handler
    return self.handlers[event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 642, in _handle_eio_message
    self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 532, in _handle_connect
    success = self._trigger_event(
  File "/home/n/anaconda3/lib/python3.8/site-packages/socketio/server.py", line 609, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/home/n/anaconda3/lib/python3.8/site-packages/flask_socketio/__init__.py", line 825, in _handle_event
    ret = handler()
  File "/home/n/Documents/Python/pvserv/run_flask.py", line 19, in handle_connect
    ensure_directory(app)
NameError: name 'ensure_directory' is not defined
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Detected change in '/home/n/Documents/Python/pvserv/run_flask.py', reloading
INFO:werkzeug: * Restarting with watchdog (inotify)
INFO:werkzeug: * Restarting with watchdog (inotify)
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 525-279-433
